z-index refactoring

Simon-Laux opened the pull request:
remove z-index where it was not needed, where it was needed tried to convert it to a lower or local/scoped z-index and put the remaining global ones into one file (`_variables.scss`).

## Commit messages
- scope the FAB z-index to the chatlist to get rid of its global z-index
- remove unessesary z-index from .group-image-edit-button
- reduce dependency on z-index in qr scanner ui
- remove dependence on z-index from fullscreen media view
- rm z-index from connectivity toast
- rm unnecessary z-indexes in jump-down-button
- put all global z-index values into a scss map in _variables.scss
- fix fmt

Simon-Laux commented:
ci failure is unrelated to change, most likely sth wrong with the npm package for the core/jsonrpc types or ci caching issue because it complains about the recently changed device message api.

ralphtheninja commented on scss/_global.scss:
```diff
@@ -144,12 +144,13 @@ code {
   image-rendering: pixelated;
 }
 
+// toast for errors and success messages
```
Cheers! :beers: :smile:

ralphtheninja commented on scss/chat/_chat-list-item.scss:
```diff
@@ -327,7 +327,7 @@
   --iconSize: 20px;
   position: fixed;
-  z-index: 20;
+  z-index: 1; // local z-index scoped to the chatlist
```
You could also move these to the variables file. Name the key accordingly to the comment, e.g. `'local-scope-chatlist'` or similar. This way we don't need these comments and we really have all z-index values in one file.

ralphtheninja commented on scss/dialogs/_qr-code.scss:
```diff
@@ -27,7 +27,7 @@
 .scan-qr-description {
   position: relative;
-  z-index: 10;
+  z-index: 1;
```
Same here. This is also a locally scoped z-index?

Simon-Laux replied:
no it isn't not really necessary with those z-index 1 values, but sure for completeness sake I can do it.

ralphtheninja commented on scss/message/_message-list.scss:
```diff
@@ -23,11 +23,10 @@
     .counter {
       position: relative;
-      z-index: 10;
+      z-index: 1;
```
Ditto.

ralphtheninja commented on src/renderer/components/AbsolutePositioningHelper/styles.module.scss:
```diff
@@ -1,4 +1,6 @@
 .absolutePositioningHelper {
   position: fixed;
-  z-index: 1000;
+  z-index: map-get($z-index, absolutePositioningHelper);
```
I'd use the same convention for all keys, e.g. `'absolute-positioning-helper'`. I realize this class isn't named the same way, but we can change this later.

ralphtheninja reviewed (commented):
Good that we got rid of some z-index values that weren't needed. But maybe consider moving all the ones that are needed to the variables file, now we're sort of in the middle of not centralizing the z-index values and also spreading some out.

Note that the key values for the z-index map doesn't _have_ to be named as the css class. They could be given different meaning in the key string. See my comments.

ralphtheninja commented on scss/chat/_chat-list.scss:
```diff
@@ -1,4 +1,5 @@
 .chat-list {
+  z-index: 0; // this scopes the z-index of the floating action button to the chatlist
```
Same here

Simon-Laux commented:
moved all into the map, added missing local scopes and made an extra sass var for creating new local scopes: `$z-index-new-local-scope`.
I also removed the `z-index` of the map overlay, as it could also be solved by reordering the html elements.

Simon-Laux commented:
No time to wait for a review, I tested it myself and self-reviewed the code to ensure that it does not break anything obvious.

ralphtheninja commented:
Good job!
